src/common/utf8.c

```diff
@@ -107,52 +135,69 @@ unsigned long decode_utf8(unsigned char *buf, int nbytes)
 
 int check_utf8(const char *buf, int len)
 {
-	unsigned char u[MAX_UTF8_SZ];
-	int enc_len = 0;
-	int i = 0;
-	while (1) {
-		unsigned int c = buf[i];
-		if (i >= len || c < 0x80 || (c & 0xC0) != 0x80) {
-			// the start of a new character. Process what we have
-			// in the buffer.
-			if (enc_len > 0) {
-				int re_encoded_len;
-				unsigned char re_encoded[MAX_UTF8_SZ];
-				unsigned long code = decode_utf8(u, enc_len);
-				if (code == INVALID_UTF8_CHAR) {
-					//printf("decoded to invalid utf8");
-					return i + 1;
+	/*
+	 * "char" is "signed" on x86 but "unsigned" on aarch64 by default.
+	 * Below code depends on signed/unsigned comparisons, define an
+	 * unsigned buffer explicitly to fix the gap.
+	 */
+	const unsigned char *bufu = (const unsigned char *)buf;
```

Comment:
nit, could use `auto` here

Reply:
C++ feature? Maybe not applicable to this C file.